llama-index-core/llama_index/core/indices/document_summary/base.py

```diff
@@ -199,11 +199,13 @@ def _add_nodes_to_index(
                 nodes=nodes_with_scores,
             )
             summary_response = cast(Response, summary_response)
+            doc_metadata = {} if not doc_id_to_nodes[doc_id] else doc_id_to_nodes[doc_id][0].metadata
```

Comment:
isn't it safer to do `doc_metadata = doc_id_to_nodes[doc_id][0].metadata if doc_id in doc_id_to_nodes else {}`

Reply:
you probably meant , cleaner , but the previous code in the function assign that doc_id in that doc, its a useless if to, the check here is to know the 0 element in the array

Reply:
your thoughts?

Reply:
your thoughts? i can choose your version either way, but there is scenario both , id check and 0 element is nessacary then or not because its self declared in the code before:
<img width="713" alt="Screenshot 2024-03-06 at 01 52 52" src="https://github.com/run-llama/llama_index/assets/17707212/bf56732d-1c18-4072-bf03-d6a06ced35e1">


so it could be :
```python doc_metadata = doc_id_to_nodes[doc_id][0].metadata ```